src/openms/include/OpenMS/FORMAT/HANDLERS/XQuestResultXMLHandler.h

```diff
@@ -0,0 +1,169 @@
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Lukas Zimmermann $
+// $Authors: Lukas Zimmermann $
+// --------------------------------------------------------------------------
+#ifndef OPENMS_FORMAT_HANDLERS_XQUESTRESULTXMLHANDLER_H
+#define OPENMS_FORMAT_HANDLERS_XQUESTRESULTXMLHANDLER_H
+
+#include <OpenMS/FORMAT/HANDLERS/XMLHandler.h>
+#include <OpenMS/METADATA/MetaInfoInterface.h>
+#include <OpenMS/METADATA/XQuestResultMeta.h>
+#include <OpenMS/METADATA/PeptideIdentification.h>
+#include <OpenMS/METADATA/ProteinIdentification.h>
+#include <OpenMS/METADATA/PeptideHit.h>
+#include <OpenMS/CHEMISTRY/EnzymesDB.h>
+
+namespace OpenMS
+{
+  namespace Internal
+  {
+    /** @brief XMLHandler for the result files of XQuest
+     */
+    class OPENMS_DLLAPI XQuestResultXMLHandler :
+      public XMLHandler
+    {
+    public:
+
+      // Maps enzyme_num in xQuest result file to the enzyme name used by OpenMS
+      static std::map< Size, String > enzymes;
+
+      // Maps String encoding month to the numeric value
+      static std::map<String, UInt> months;
+
+      // Decoy string used by xQuest
+      static const String decoy_string;
+
+      /**
+       * @brief Removes a substring from a larger string
+       * @param large String from which `small` should be removed
+       * @param small Substring to be removed from `large`
+       */
+      static void removeSubstring(String & large, const String & small);
+
+      XQuestResultXMLHandler(const String & /* filename */,
+                             std::vector< XQuestResultMeta > & /* metas */,
+                             std::vector< std::vector< PeptideIdentification > > & /* csms */,
+                             std::vector< ProteinIdentification > &,
+                             int & n_hits,
+                             double & min_score,
+                             double & max_score,
+                             std::vector< int > * cum_hits,
+                             Size min_n_ions_per_spectrum,
+                             bool load_to_peptideHit_);
+      virtual ~XQuestResultXMLHandler();
+
+      // Docu in base class
+      void endElement(const XMLCh * const /*uri*/, const XMLCh * const /*local_name*/, const XMLCh * const qname);
+
+      // Docu in base class
+      void startElement(const XMLCh * const /*uri*/, const XMLCh * const /*local_name*/, const XMLCh * const qname, const xercesc::Attributes & attributes);
+
+      // Docu in base class
+      //void characters(const XMLCh * const chars, const XMLSize_t /*length*/);
+
+      //Docu in base class
+      //virtual void writeTo(std::ostream & os);
+
+    private:
+   
+      // Main data structures that are populated during loading the file
+      std::vector< XQuestResultMeta > & metas_;
```

Comment:
Just a side note: I think for a file handler it is fine to use references in members as it doesn't provide many possibilities for refactorings anyways. Otherwise, the use of & for member variables is discouraged as this introduces a (potentially unneeded) dependency inside a class.